# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/grub2/hotfix-no-zicsr.patch.riscv
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- grub-2.06/grub-core/kern/riscv/cache_flush.S.vanilla	2022-11-26 23:19:31.446668490 +0100
+++ grub-2.06/grub-core/kern/riscv/cache_flush.S	2022-11-26 23:20:01.163665855 +0100
@@ -40,5 +40,7 @@
  * a2 - line size
  */
 FUNCTION(grub_arch_invalidate_icache_range)
+#ifdef __riscv_zicsr
 	fence.i
+#endif
 	ret
